packages/api/src/resolvers/user/index.ts

```diff
@@ -355,3 +361,47 @@ export const signupResolver: ResolverFn<
     return { errorCodes: [SignupErrorCode.Unknown] }
   }
 }
+
+export const deleteAccountResolver = authorized<
+  DeleteAccountSuccess,
+  DeleteAccountError,
+  MutationDeleteAccountArgs
+>(async (_, { userID }, { models, claims, log, authTrx, pubsub }) => {
+  const user = await models.user.get(userID)
+
+  if (!user) {
+    return {
+      errorCodes: [DeleteAccountErrorCode.UserNotFound],
+    }
+  }
+
+  if (user.id !== claims.uid) {
+    return {
+      errorCodes: [DeleteAccountErrorCode.Unauthorized],
+    }
+  }
+
+  const deleteUserResult = await authTrx((tx) =>
+    models.user.deleteUser(claims.uid, tx)
+  )
+
+  // delete this user's pages in elastic
+  await deletePagesByParam({ userId: userID }, { uid: userID, pubsub })
+
+  log.info('Deleting a user account', {
+    userID,
+    labels: {
+      source: 'resolver',
+      resolver: 'deleteAccountResolver',
+      uid: claims.uid,
+    },
+  })
+
+  if ((deleteUserResult as UserData).id !== undefined) {
+    return { userID }
+  } else {
+    return {
+      errorCodes: [DeleteAccountErrorCode.Forbidden],
```

Comment:
@sywhb Tried to delete an account a few times but it looks like it's ending up here and sending a FORBIDDEN error code. Doesn't seem to be a type error either since the account is still intact when I logout and in again 🤔 

Do you mind checking the logs to see if you find anything? I don't think I have access for them. Only have access to Sentry which doesn't have anything helpful here.

Reply:
<img width="952" alt="Screen Shot 2022-07-07 at 8 59 35 AM" src="https://user-images.githubusercontent.com/836745/177818834-4fa9dad8-367b-48dc-9c01-96ec2a521358.png">

Reply:
@satindar cool. I will see what I can find

Reply:
@satindar finally i figured out the issue is with the db permission. fixed it and deploying to demo

Reply:
@satindar deployed on demo! verified in my local env. let me know if it works :)